File: queue.cpp
#include <iostream>
#include <string>

using namespace std;

#define MAX 5

class Queue 
{
    private:
    int arr [MAX];
    int front, rear;

    public:
    Queue() {
        front = -1;
        rear = -1;
    }

    //enqueue op.
    void enqueue(int value) 
    {
        if (rear == MAX-1)
        {
            cout << "Queue Overflow." << endl;
            return;
        }
        if (front == -1)
        front = 0;
        rear ++;
        arr[rear] = value;
        cout << value << " Inserted into the queue." <<endl;
    }

    void dequeue()
    {
        if (front == -1 || front > rear)
        {
            cout << "Queue Underflow. queue is empty." << endl;
            return;
        }
        cout << arr[front] << " Delected from Queue." << endl;
        front ++;
        if (front > rear) 
        {
            front = rear = -1;
        }
    }

    void display()
    {
        if (front == -1) 
        {
            cout << "Nothing to display." << endl;
            return;
        }
        cout  << "Queue elements:";
        for (int i = front; i <=rear; i++)
        {
            cout << arr [i] << " ";
        }
        cout << endl;
    }

};

int main() {
    Queue q;
    int choice, value;

    do {
        cout <<"\n ---- Queue operations ----";
        cout <<"\n 1.Enqueue";
        cout <<"\n 2.Dequeue";
        cout <<"\n 3.Display";
        cout <<"\n 4.Exit program";
        cout <<"\n Enter choice:    ";
        cin >> choice;

        switch (choice)
        {
            case 1:
                cout  << "Enter value to insert:    ";
                cin >> value;
                q.enqueue (value);
                break;

            case 2:
                q.dequeue ();
                break;

            case 3:
                q.display();
                break;

            case 4:
                cout << "bye!" << endl;
                break;

            default:
                cout << "Invalid choice! try again :P" << endl;

        }

    } while (choice != 4);
    return 0;
};

File: PythonCodes/Queue.py
#creating a fixed size queue using arrays

class fixedqueue:
    def __init__(self, capacity: int):
        self.capacity = capacity
        self.queue = [None] * capacity
        self.front = 0
        self.end = 0
        self.size = 0

    def isempty(self) -> bool:
        return self.size == 0

    def isfull(self) -> bool:
        return self.size == self.capacity

    def enqueue(self, item):
        if self.isfull():
            print ("Queue Overflow!!")

        else:
            self.queue[self.end] = item
            self.end += 1
            self.size +=1
            

    def dequeue(self):
        if self.isempty():
            print ("Queue Underflow!!")

        else:
            item = self.queue[self.front]
            self.queue[self.front] = None
            self.front += 1
            self.size -= 1
            return item

    def peek(self):
        if self.isempty():
            print ("Queue is empty! nothing to see.")

        else:
            print ("Front element   :   ", self.queue[self.front])

    def display(self):
        if self.isempty():
            print ("Queue is empty! nothing to see.")

        else:
            active_elements = self.queue[self.front : self.end + 1]
            print ("Your Queue     :   ", active_elements)



if  __name__ == "__main__":
    capacity = int(input("Enter queue capacity  :"))
    queue = fixedqueue(capacity)

    while True:
        print ("----Queue Menu----")
        print ("1.Enqueue \n2.Dequeue \n3.Peek \n4.Display \n5.Exit")

        choice = int(input("Enter your choice   :"))

        if choice == 1:
            item = (input("Enter element to push    :"))
            queue.enqueue(item)

        elif choice == 2:
            popped = queue.dequeue()

            print ("Item popped :", popped)

        elif choice == 3:
            queue.peek()

        elif choice == 4:
            queue.display()

        elif choice == 5:
            print("Byee!!")
            break

        else:
            print("Invalid Try again!")
